webgrid/testing.py

```diff
@@ -343,8 +360,9 @@ def get_session_grid(self, *args, **kwargs):
             grid instance
         """
         grid = self.grid_cls(*args, **kwargs)
-        grid.apply_qs_args()
-        return grid
+        with grid.manager.request_context(url=self.url):
```

Comment:
If the surrounding test has set up a request context already, we should preserve that here. Need to check the stack to see if one has been pushed.